src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,134 @@
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+type PostProps = {
+  postId: number,
+  content: string,
+};
+
+export const PostDetails: React.FC <PostProps> = ({
+  postId,
+  content,
+}) => {
+  // alert('Post details rendering');
+
+  const [
+    areCommentsVisible,
+    setAreCommentsVisible,
+  ] = useState(true);
+
+  const [
+    comments,
+    setComments,
+  ] = useState<Comment[]>([]);
+
+  const [
+    commentsChanged,
+    setCommentsChanged,
+  ] = useState(false);
+
+  const toggleCommentsVisibility = () => {
+    setAreCommentsVisible(prevState => !prevState);
+  };
+
+  const addNewComment = (
+    event: React.FormEvent<HTMLFormElement>,
+    currentPostId: number,
+    commentUserName: string,
+    commentEmail: string,
+    commentText: string,
+  ) => {
+    event.preventDefault();
+
+    addComment(currentPostId, commentUserName, commentEmail, commentText);
+    setCommentsChanged(prevState => !prevState);
+  };
+
+  const removeComment = (commentId: number) => {
+    deleteComment(commentId);
+    setCommentsChanged(prevState => !prevState);
+  };
+
+  useEffect(() => {
+    getPostComments(postId).then(commentsArr => setComments(commentsArr));
+  }, []);
+
+  useMemo(async () => {
+    const commentsArr = await getPostComments(postId);
+
+    setComments(commentsArr);
+  }, [commentsChanged, postId]);
+
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+
+      <section className="PostDetails__post">
+        <p>
+          {content}
+        </p>
+      </section>
+
+      <section className="PostDetails__comments">
+
+        {
+          (comments.length > 0)
+          && (
+            <button
+              type="button"
+              className="button"
+              onClick={toggleCommentsVisibility}
+            >
+              {
+                areCommentsVisible
+                  ? ` Hide ${comments.length} comments`
+                  : ` Show ${comments.length} comments`
+              }
+            </button>
+          )
+        }
+
+        {
+          areCommentsVisible
+            && (
+              <ul className="PostDetails__list">
+                {
+                  comments.map(comment => (
+                    <>
```

Comment:
remove fragment, cause if you use it you need to add key to the fragment itself, not to li
(but I think fragment is not needed here. so better remove it at all)

Reply:
Done